### PR TITLE
[Fix][API] Fix get SeaTunnelRow size failed when array value has `null`

### LINKED ISSUE #6455: [Bug] [SinkMetrics] the sink metrics collect will get NPE when data type is String[]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

[SinkFlowLifeCycle#250 ](https://github.com/apache/seatunnel/blob/dev/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/flow/SinkFlowLifeCycle.java#L251)
https://github.com/apache/seatunnel/blob/dev/seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SeaTunnelRow.java#L240

in sink life cycle, will summary the sink bytes. but if the data type is string array, doesn't have null check, will get NPE when value is null.

### SeaTunnel Version

2.3.4

### SeaTunnel Config

```conf
na
```

### Running Command

```shell
na
```

### Error Exception

```log
NPE
```

### Zeta or Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fcb-xiaobo**
### Purpose of this pull request
when get output data byte  size info,There is null data in the collection, abnormal length obtained

### CODE REVIEW

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SeaTunnelRow.java [hunk 1]
```diff
@@ -237,7 +237,9 @@ private int getBytesForValue(Object v) {
             case "String[]":
                 int s = 0;
                 for (String i : ((String[]) v)) {
-                    s += i.length();
+                    if (null != i) {
+                        s += i.length();
+                    }
                 }
                 return s;
             case "Boolean[]":
```
Comment (Hisoka-X) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SeaTunnelRow.java]: Thanks @fcb-xiaobo ! I think we need to abstract a general method to filter out null values ​​in all types of arrays. Not only `String[]` but `Boolean[]`,`Long[]` etc.

**Review by Hisoka-X — COMMENTED** [review]
Also please add test in https://github.com/apache/seatunnel/blob/6ec16ac46fe8b36c733556f100ac7bb7e4a7e9f4/seatunnel-api/src/test/java/org/apache/seatunnel/api/table/type/SeaTunnelRowTest.java#L29

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SeaTunnelRow.java (hunk 1 above)
Comment (fcb-xiaobo) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SeaTunnelRow.java]: ok，i try abstract common methon check all values is null

**Comment by liunaijie** [comment]
#6455

File: seatunnel-api/src/test/java/org/apache/seatunnel/api/table/type/SeaTunnelRowTest.java [hunk 2]
```diff
@@ -104,4 +104,23 @@ void testWithMapInterface() {
...
+        Assertions.assertEquals(16, doubleRow.getBytesSize());
+
+        String str = "seatunnel";
+        SeaTunnelRow strRow = new SeaTunnelRow(new Object[] {str});
+        Assertions.assertEquals(9, strRow.getBytesSize());
+
+        String nonStr = null;
+        SeaTunnelRow nonStrRow = new SeaTunnelRow(new Object[] {nonStr});
+        Assertions.assertEquals(0, nonStrRow.getBytesSize());
+    }
```
Comment (Hisoka-X) [inline seatunnel-api/src/test/java/org/apache/seatunnel/api/table/type/SeaTunnelRowTest.java:125]: Please add test case to cover `Long[]`, `Double[]`,`String[]` etc. All array type.

**Comment by fcb-xiaobo** [comment]
Sorry, I'm a little busy at work recently and don't have time to deal with this problem, so I'll close it first.

**Pull request closed without merging**